### DISCUSSION TITLE
$nextTick() does not behave as expected while CSS animations are playing

### DISCUSSION THREAD

**Discussion by magnusskare** [alpinejs/alpine#3012] [7. Bugs (archived)] [answered]
## Summary
I'm using x-show to hide/show a form, and x-ref to move focus to the form's only input element. In order to make sure the form is visible on the page before focusing, I'm using `$nextTick()`. In Chrome, this does not work as expected if there are CSS animations playing or waiting to be played (e.g. with `animation-delay`). It seems to be related to the combination of `x-show` and `$nextTick()`, and affects browsers in different ways.

**Expected behavior:** The input should receive focus when the button or keyboard shortcut is pressed, even when CSS animations are playing.

**Actual behavior:** The input does not receive focus consistently when CSS animations are playing. When the animation finishes, the focus-trigger behaves as expected.

**Reduced example:** [Codepen](https://codepen.io/partcoffee/pen/vYRYQvN)

## Affected browsers for the test:
* Chrome: Works, but only after the CSS animation ends
* Safari: Does not work at all, but seems to be related to the timing of x-show and $nextTick
* Firefox: Works with whatever you throw at it

#### Show input with x-show and move focus with $nextTick
[Reduced test](https://codepen.io/partcoffee/pen/VwXwqbK)
* Chrome: Works
* Safari: Does not work
* Firefox: Works

#### Move focus with $nextTick (form is visible all the time)
[Reduced test](https://codepen.io/partcoffee/pen/BaravdY)
* Chrome: Works
* Safari: Works
* Firefox: Works

**Comment by magnusskare** [comment]
Update: I did a quick search, and the Safari behavior seems related to this issue #2855 — so consider the Safari issue explained. The Chrome + animation-bug seems unrelated to this.

**Comment by SimoTod** [comment, marked as answer]
x-show is internally delayed to be compatible with click.away. I think you need to wait for 2 ticks for it to work properly

**Reply by magnusskare** [reply]
Thanks, you put me on the right track. Ended up using a timeout, as I'm just triggering the focus state.
